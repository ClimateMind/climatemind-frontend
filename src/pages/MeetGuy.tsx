import React from 'react';
import { useHistory } from 'react-router-dom';
import { Typography, Button, Grid, makeStyles, Box } from '@material-ui/core';
import { ReactComponent as Logo } from '../assets/cm-logo.svg';
import ROUTES from '../components/Router/RouteConfig';

const styles = makeStyles({
  root: {
    flexGrow: 1,
    backgroundColor: '#FF9439',
    minHeight: '100vh',
  },
  typography: {
    textAlign: 'center',
  },
});

const MeetGuy: React.FC<{}> = () => {
  const classes = styles();
  const history = useHistory();

  return (
    <Grid container className={classes.root} data-testid="MeetGuy">
      <Grid item sm={false} lg={4}>
        {/* left gutter */}
      </Grid>

      <Grid item sm={12} lg={4}>
        <Grid container direction="column" alignItems="center">
          <Grid item>
            <Box mt={12} mb={14}>
              <Typography variant="h4">Hello there!</Typography>
            </Box>
          </Grid>

          <Grid item>
            <Typography variant="h6">Welcome to Climate Mind.</Typography>
          </Grid>

          <Grid item>
            <Box mt={4} mb={10}>
              <Logo data-testid="climate-mind-logo" />
            </Box>
          </Grid>

          <Grid item>
            <Box pr={10} pl={10}>
              <Typography className={classes.typography}>
                Iâ€™ll help you find out your Climate Personality to give you
                personalised solutions to climate change.
              </Typography>
            </Box>
          </Grid>

          <Grid item>
            <Box mt={4} mb={8}>
              <Button
                variant="contained"
                color="primary"
                disableElevation
                onClick={() => history.push(ROUTES.ROUTE_PERSONALITY)}
              >
                Let's Go!
              </Button>
            </Box>
          </Grid>
        </Grid>
      </Grid>

      <Grid item sm={false} lg={4}>
        {/* right gutter */}
      </Grid>
    </Grid>
  );
};

export default MeetGuy;
