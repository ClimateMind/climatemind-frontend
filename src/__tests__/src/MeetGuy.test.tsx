import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import MeetGuy from '../../pages/MeetGuy';

// Mock react router to simulate history.push on button click
jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe('Quiz Welcome', () => {
  it('has a get started button which has be clicked', () => {
    const { getByRole } = render(<MeetGuy />);
    expect(getByRole('button'));
    fireEvent(
      getByRole('button'),
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      })
    );
  });

  it('displays the app logo', () => {
    const { getByTestId } = render(<MeetGuy />);
    expect(getByTestId('climate-mind-logo')).toBeInTheDocument();
  });
  it('shows welcome text ', () => {
    const { getByText } = render(<MeetGuy />);
    expect(getByText(/find out your Climate Personality/i)).toBeInTheDocument();
  });
  it('Matches the snapshot', () => {
    const { getByTestId } = render(<MeetGuy />);
    expect(getByTestId('MeetGuy')).toMatchSnapshot();
  });
});
