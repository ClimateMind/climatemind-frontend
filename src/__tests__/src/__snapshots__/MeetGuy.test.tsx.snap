// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Welcome Matches the snapshot 1`] = `
<div
  class="MuiGrid-root makeStyles-root-19 MuiGrid-container"
  data-testid="MeetGuy"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-4"
  />
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-lg-4"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiBox-root MuiBox-root-21"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
            Hello there!
          </h4>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Welcome to Climate Mind.
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiBox-root MuiBox-root-22"
        >
          <svg
            data-testid="climate-mind-logo"
          >
            cm-logo.svg
          </svg>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiBox-root MuiBox-root-23"
        >
          <p
            class="MuiTypography-root makeStyles-typography-20 MuiTypography-body1"
          >
            Iâ€™ll help you find out your Climate Personality to give you personalised solutions to climate change.
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiBox-root MuiBox-root-24"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Let's Go!
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-4"
  />
</div>
`;
